snapcraft/internal/project_loader/grammar/_processor.py

```diff
@@ -15,8 +15,11 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 import re
+from typing import Any, Callable, Dict, List, Set
 
+from snapcraft import project
 from .errors import GrammarSyntaxError
+from ._statement import Statement, GrammarType, CallStackType
```

Comment:
maybe this should rather be
```python
from . import typing
```

and then use it like

```python
def __init__(self, grammar: typing.Grammar, ...):
```

To continue with the flow of current native types.

Reply:
Ah, I like it!

Reply:
Done.